src/core/unit.h

```diff
@@ -147,6 +147,7 @@ typedef struct Unit {
 
         /* The slot used for watching NameOwnerChanged signals */
         sd_bus_slot *match_bus_slot;
+        char *match_bus_name;
```

Comment:
storing this should not be necessary if you enqueue the GetNameOwner call right-away

Reply:
Hmm, in the GetNameOwner callback I still need the name and does not have access to it...

Reply:
hmm, i figure we can just drop the name from the Unit's name_owner_change callbcak prototype... we don't really neeed that. In the one implementor (service.c) we just check that it matches what we thinkg it does, but that's not useful. Hence maybe just get rid of it? should simplify stuff.

Reply:
As I noted as in comment on the PR, there is code that assumes that service is the only user of name_owner_change watches, what do you think of moving it all into the service class. It certainly makes things a lot more accessible...